driver-core/src/main/java/com/datastax/driver/core/JdkSSLOptions.java

```diff
@@ -0,0 +1,131 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.security.NoSuchAlgorithmException;
+
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.ssl.SslHandler;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+
+/**
+ * {@link SSLOptions} implementation based on built-in JDK classes.
+ */
+public class JdkSSLOptions implements SSLOptions {
+
+    /**
+     * Creates a builder to create a new instance.
+     *
+     * @return the builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    private final SSLContext context;
+    private final String[] cipherSuites;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param context the SSL context.
+     * @param cipherSuites the cipher suites to use.
+     */
+    protected JdkSSLOptions(SSLContext context, String[] cipherSuites) {
+        this.context = (context == null) ? makeDefaultContext() : context;
+        this.cipherSuites = cipherSuites;
+    }
+
+    @Override
+    public SslHandler newSSLHandler(SocketChannel channel) {
+        SSLEngine engine = newSSLEngine(channel);
+        return new SslHandler(engine);
+    }
+
+    /**
+     * Creates an SSL engine each time a connection is established.
+     *
+     * <p>
+     * You might want to override this if you need to fine-tune the engine's configuration
+     * (for example enabling hostname verification).
+     *
+     * @param channel the Netty channel for that connection.
+     * @return the engine.
+     */
+    protected SSLEngine newSSLEngine(SocketChannel channel) {
```

Comment:
It's not used in this default implementation, but it can be useful if you override the method to do something else. In particular, that's how you would get access to the remote address and port.